### ISSUE TITLE
NestJS app cannot start when a remote SOAP service is unavailable

### ISSUE THREAD

**Issue by vmanchev** [lehh/nestjs-soap#16]
Hello,
and thank you for your work, its appreciated!

I am using `nestjs-soap` (version 1) to call an external, 3rd party SOAP service. The problem is, this 3rd party SOAP service is not always available, its often down for maintenance. In such moment, my NestJS application will not start. It throws an error like:

```
[Nest] 3548   - 02/05/2022, 6:52:27 PM   [ExceptionHandler] write EPROTO 16936:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:c:\ws\deps\openssl\openssl\ssl\record\ssl3_record.c:332:
 +716ms
Error: write EPROTO 16936:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:c:\ws\deps\openssl\openssl\ssl\record\ssl3_record.c:332:

    at D:\projects\ven\vet-api\node_modules\nestjs-soap\dist\soap-utils.js:19:19
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I've tracked this down to:

```
throw new common_1.ServiceUnavailableException(err);
```

This is the way I register the `SoapModule`

```
SoapModule.registerAsync([{ 
    name: 'VAT_SOAP_CLIENT', 
    uri: 'https://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl' 
}]),
```

As far as I understood, the moment we import the module, it tries to connect to that service immediately. Well, if my NestJS app is already started and the SOAP service is unavailable at some moment, that is fine, I can handle this case. 

The problem is when the SOAP service is unavailable when the NestJS app is build and I try to start it. It wouldn't start. No matter the way we initialize the `SoapModule`, we just pass a config object and it tries to connect immediately. 

Is there a way to postpone the connection to the SOAP service to the moment where we actually want to make a call, not at the moment when the module is registered?

**Comment by lehh** [comment]
Hello @vmanchev,

Thanks for bringing this up!

I could change the module initialisation to just log the error when the client is not created successfully, instead of throwing it and crashing the application. This way, you will need to check whether the client is null when trying to use it. 
Example:
```typescript
export class ExampleService {
  constructor(@Inject('MY_SOAP_CLIENT') private readonly mySoapClient: Client) {}

  async exampleFunction() {
   if (!this.mySoapClient)
      return "Service Unavailable";

    return await this.mySoapClient.YourFunctionAsync();
  }
}
```

What do you think of this approach?

**Comment by vmanchev** [comment]
@lehh this will be perfect!

**Comment by lehh** [comment]
@vmanchev All right. 

I'll have to make this change in v2 because v1 is deprecated and I'm not maintaining it anymore.

It's easy to migrate to v2. You can check the [upgrading to v2 docs](https://github.com/lehh/nestjs-soap/blob/main/docs/upgrading-to-v2.md).

**Comment by lehh** [comment]
Released on [v2.0.1](https://github.com/lehh/nestjs-soap/releases/tag/v2.0.1) :)

**Comment by vmanchev** [comment]
Thanks a lot! It works like a charm!
